core/src/jvmMain/kotlin/com.saveourtool.processbuilder/ProcessBuilderInternal.kt

```diff
@@ -13,70 +13,72 @@ import okio.Path
 import java.io.BufferedReader
 import java.io.InputStreamReader
 
+import kotlin.time.Duration
 import kotlinx.coroutines.DelicateCoroutinesApi
 import kotlinx.coroutines.delay
 import kotlinx.coroutines.launch
 import kotlinx.coroutines.newFixedThreadPoolContext
 import kotlinx.coroutines.runBlocking
 
 actual class ProcessBuilderInternal actual constructor(
-    private val stdoutFile: Path,
-    private val stderrFile: Path,
-    private val useInternalRedirections: Boolean,
+    private val redirects: ProcessBuilderConfig.Redirects,
+    private val childProcessUserName: String?,
 ) {
-    actual fun prepareCmd(command: String): String {
-        val cmd = buildList {
-            if (isCurrentOsWindows()) {
-                add("CMD")
-                add("/C")
-            } else {
-                add("sh")
-                add("-c")
-            }
-            add(command)
+    actual fun appendShellCommand(command: String): String = buildList {
+        if (isCurrentOsWindows()) {
+            add("CMD")
```

Comment:
1. Please change it to `cmd.exe` -- otherwise, `cmd.bat`, `cmd.com`, or `cmd.cmd` can be launched first if they're in the `PATH`.
2. Please add a new task to the issues: cmd.exe should either be taken from %COMSPEC%, or searched in a limited set of well-known locations if `COMSPEC` is empty. For security reasons, you shouldn't launch an arbitrary `cmd.exe` found in `PATH`.